src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -308,7 +325,8 @@ public static ModelMetadata fromString(String modelMetadataString) {
                 description,
                 error,
                 "",
-                MethodComponentContext.EMPTY
+                MethodComponentContext.EMPTY,
+                VectorDataType.FLOAT
```

Comment:
@junqiu-lei Why are we trying to set it as FLOAT instead of fetching it from metadata array ?

Reply:
Understood why you are passing FLOAT here. But, do we really need to add a separate if else block for vectordatatype because it is an optional field. Can we include this field in existing if else blocks ?

Reply:
NVM..saw Jack and Ryan's comment about BWC